library/init/meta/expr_address.lean

```diff
@@ -0,0 +1,86 @@
+prelude
+import init.meta.expr
+import init.data.list.basic
+import init.data.option.basic
+
+namespace expr
+
+/-- An enum representing a recursive argument in an `expr` constructor (except the types of variables). -/
+inductive coord : Type
+| app_fn        | app_arg
+| lam_var_type  | lam_body
+| pi_var_type   | pi_body
+| elet_var_type | elet_assignment | elet_body
+
+namespace coord
+
+def code: coord → ℕ
+| coord.app_fn        := 0 | coord.app_arg         := 1
+| coord.lam_var_type  := 2 | coord.lam_body        := 3
+| coord.pi_var_type   := 4 | coord.pi_body         := 5
+| coord.elet_var_type := 6 | coord.elet_assignment := 7 | coord.elet_body := 8
+
+def repr: coord → string
+| coord.app_fn        := "app_fn"        | coord.app_arg  := "app_arg"
+| coord.lam_var_type  := "lam_var_type"  | coord.lam_body := "lam_body"
+| coord.pi_var_type   := "pi_var_type"   | coord.pi_body := "pi_body"
+| coord.elet_var_type := "elet_var_type" | coord.elet_assignment := "elet_assignment" | coord.elet_body := "elet_body"
+
+instance : has_repr coord := ⟨repr⟩
```

Comment:
Please also add instances for the other stringification classes: `has_to_string` and `has_to_format` (`has_to_tactic_format` is implied by `has_to_format`).